packages/next/src/shared/lib/constants.ts

```diff
@@ -54,7 +54,11 @@ export const DEV_MIDDLEWARE_MANIFEST = '_devMiddlewareManifest.json'
 export const REACT_LOADABLE_MANIFEST = 'react-loadable-manifest.json'
 export const FONT_MANIFEST = 'font-manifest.json'
 export const SERVER_DIRECTORY = 'server'
-export const CONFIG_FILES = ['next.config.js', 'next.config.mjs']
+export const CONFIG_FILES = [
+  'next.config.js',
+  'next.config.mjs',
+  'next.config.ts',
+]
```

Comment:
Since we are adding `next.config.ts`, why not also `next.config.mts` and `next.config.cts`?

Reply:
Adding `cts` and `mts` looks unnecessary, but I'll add to read off from `tsconfig` `compilerOptions.module` to specify the format and the output.

Reply:
Specifying `mts` and `cts` is precisely what people would do if they want a file to be opted out of the specified module format in the `tsconfig.json`.

Reply:
E.g. if I have specified my module format to `esnext` in the `tsconfig.json`, but I want to require a CommonJS module in the next config, I would expect `next.config.cts` to work.

Reply:
For this PR I'll concentrate on `next.config.ts`, and we can add more variations later on if needed.

Reply:
@SukkaW I've modified to transform as `cjs` format regardless of the source, and I guess it'd also resolve your suggestion!

P.S. Added more tests, feel free to give feedback